src/transformStateWithClones.js

```diff
@@ -7,7 +7,26 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let currentState = { ...state };
```

Comment:
[NAMING] - The name 'currentState' can be misleading as it is used to create a new state object that will be pushed to the stateHistory. A better name could be 'stateClone' or 'nextState' to reflect that it is a new state derived from the previous one.